traffic_ops/testing/api/v4/parameters_test.go

```
@@ -246,3 +287,290 @@ func DeleteTestParameter(t *testing.T, pl tc.Parameter) {
+		if reqInf.StatusCode != http.StatusBadRequest {
+			t.Errorf("Expected 400 status code, got %v", reqInf.StatusCode)
+		}
+	}
+}
+
+func CreateTestParametersMissingName(t *testing.T) {
+	opts := client.NewRequestOptions()
+	if len(testData.Parameters) > 0 {
+		firstParameter := testData.Parameters[0]
+		firstParameter.Name = ""
+		_, reqInf, _ := TOSession.CreateParameter(firstParameter, opts)
+		if reqInf.StatusCode != http.StatusBadRequest {
+			t.Errorf("Expected 400 status code, got %v", reqInf.StatusCode)
+		}
+	}
+}
+
+func CreateTestParametersMissingconfigFile(t *testing.T) {
+	opts := client.NewRequestOptions()
+	if len(testData.Parameters) > 0 {
+		firstParameter := testData.Parameters[0]
+		firstParameter.ConfigFile = ""
+		_, reqInf, _ := TOSession.CreateParameter(firstParameter, opts)
+		if reqInf.StatusCode != http.StatusBadRequest {
+			t.Errorf("Expected 400 status code, got %v", reqInf.StatusCode)
+		}
+	}
+}
+
+func UpdateParametersEmptyValue(t *testing.T) {
+	if len(testData.Parameters) > 0 {
+		firstParameter := testData.Parameters[0]
+		// Retrieve the Parameter by name so we can get the id for the Update
+		opts := client.NewRequestOptions()
+		opts.QueryParameters.Set("name", firstParameter.Name)
+		resp, _, err := TOSession.GetParameters(opts)
+		if err != nil {
+			t.Errorf("cannot GET Parameter by name: %v - %v", firstParameter.Name, err)
+		}
+		remoteParameter := resp.Response[0]
+		//Parameters can be updated with empty value, so no error while updating
+		remoteParameter.Value = ""
+		opts = client.NewRequestOptions()
+		var alert tc.Alerts
+		alert, reqInf, err := TOSession.UpdateParameter(remoteParameter.ID, remoteParameter, opts)
+		if err != nil {
+			t.Errorf("cannot UPDATE Parameter by id: %v - %v", err, alert)
+		}
+		if reqInf.StatusCode != http.StatusOK {
+			t.Errorf("Expected 200 status code, got %v", reqInf.StatusCode)
+		}
+	}
+}
+
+func UpdateParametersEmptyName(t *testing.T) {
+	if len(testData.Parameters) > 0 {
```

This test will silently pass if the test data is in a bad state and the test cannot proceed

---

Added failure checks